packages/xsnap/src/xsnap.js

```diff
@@ -119,17 +114,43 @@ export function xsnap(options) {
   /** @type {Promise<void>} */
   let baton = Promise.resolve();
 
+  /** @type {string} */
+  let xsWorkerType;
+  async function getWorkerType() {
```

Comment:
it's awkward to have the xsnap level know anything about workers.

Reply:
The worker type is specifically what version of xsnap is in use.

Reply:
ah. different sense of "worker"